From: Marco Nenciarini <mnencia@debian.org>
Date: Sat, 5 Mar 2011 11:19:07 +0100
Updated: Mon, 25 Nov 2013 20:11:59 -0500
Subject: split-protocols
Description: Set default protocols value as empty and enable each
 protocol in its own configuration file
---
 doc/example-config/dovecot.conf |    4 ++--
 src/config/all-settings.c       |    2 +-
 src/master/master-settings.c    |    2 +-
 3 files changed, 4 insertions(+), 4 deletions(-)

--- a/doc/example-config/dovecot.conf
+++ b/doc/example-config/dovecot.conf
@@ -20,8 +20,8 @@
 # options. The paths listed here are for configure --prefix=/usr
 # --sysconfdir=/etc --localstatedir=/var
 
-# Protocols we want to be serving.
-#protocols = imap pop3 lmtp submission
+# Enable installed protocols
+!include_try /usr/share/dovecot/protocols.d/*.protocol
 
 # A comma separated list of IPs or hosts where to listen in for connections. 
 # "*" listens in all IPv4 interfaces, "::" listens in all IPv6 interfaces.
--- a/src/master/master-settings.c
+++ b/src/master/master-settings.c
@@ -205,7 +205,7 @@
 	.state_dir = PKG_STATEDIR,
 	.libexec_dir = PKG_LIBEXECDIR,
 	.instance_name = PACKAGE,
-	.protocols = "imap pop3 lmtp",
+	.protocols = "",
 	.listen = "*, ::",
 	.ssl = "yes:no:required",
 	.default_internal_user = "dovecot",
